src/Paket.Core/Cache.fs

```diff
@@ -2,11 +2,47 @@
 
 open System
 open System.IO
+open System.Text.RegularExpressions
+open Paket.PackageSources
 
 type CacheType = 
-    | AllVersion
-    | CurrentVersiononly
+    | AllVersions
+    | CurrentVersion
 
 type Cache = 
     { Location : string
-      CacheType : CacheType }
\ No newline at end of file
+      CacheType : CacheType option }
+
+    static member Parse(line : string) =
+        let sourceRegex = Regex("cache[ ]*[\"]([^\"]*)[\"]", RegexOptions.IgnoreCase)
```

Comment:
There's no need for the first and last character class here: `"cache[ ]*\"([^\"]*)\""` should be fine.